The CoGAPS matrix factorization algorithm identifies the transcription factor AP-2alpha as a feedback mechanism from therapeutic inhibition of the EGFR network
====
Patients with oncogene driven tumors are currently treated with targeted therapeutics such as epidermal growth factor receptor (EGFR) inhibitors. The inhibited oncogenic pathway often interacts with other signaling pathways and alters predicted therapeutic response. Geenomic data from the Cancer Genome Atlas demonstrates pervasive molecular alterations to EGFR, MAPK, and PI3K signaling in previously untreated tumors. Therefore, this study uses bioinformatics algorithms to infer the complex pathway interactions that result when treating cancer cells with these common genetic alterations in the EGFR network with EGFR inhibitors. We modify the HaCaT keratinocyte cell line model of premalignancy to simulate malignancy that result in constitutive activation of EGFR, HRAS, and PI3K in a controlled genetic background. We then measured gene expression after treating modified HaCaT cells with three EGFR targeted agents (gefitinib, afatinib, and cetuximab) for 24 hours. The CoGAPS bioinformatics algorithm distinguishes a gene expression signature associated with the anticipated silencing of the EGFR network and a signature associated with unanticipated transcriptional feedback in the modified HaCaT cells. Notably, the feedback signature shows that EGFR gene expression itself increased in cells that are responsive to EGFR inhibitors. This CoGAPS signature further associates feedback with increased expression of several growth factor receptors by the AP-2 family of transcription factors. The resulting gene expression signatures for AP-2alpha are further associated with inhibition of pathways downstream of EGFR including MEK, PI3K, and mTOR in data obtained from Library of Integrated Network-Based Cellular Signatures. These data suggest that the AP-2 transcription factors are activated by feedback from inhibition of the EGFR network and subsequently regulate expression of other growth factor receptors that are known to cause EGFR inhibitor resistance. 

Details about the algorithm and these results are under review in **Fertig et al. (2016)** Code supporting these results is written for R organized using [ProjectTemplate](http://projecttemplate.net). 

Briefly, the file structure is as follows:

* **cache**: R objects stored from intermediate analysis, and used to obtain results in the paper.
* **data**: Cell survival data. Rdata objects containing raw gene expression data from the CEL files in [GEO](http://www.ncbi.nlm.nih.gov/geo/)
* **graphs**: plots generated from R code, used as Figures in the manuscript
* **munge**: code to preprocess gene expression data to useable format for analysis, subsequently stored in the cache. 
* **src**: code to obtain published results from data preprocessed according to munge and stored in the cache
* **reports**: tables generated from the analysis included in the manuscript and analyses of TCGA/LINCS data
